File: FormsAndWpfControls/vb/WinFormsUserControl.vb
Imports FormsAndWpfControls.My.Resources
Imports Xarial.XCad.Base.Attributes
Imports Xarial.XCad.UI.PropertyPage

<Title("WinForms Control")>
<Icon(GetType(Resources), NameOf(Resources.winforms_icon))>
Public Class WinFormsUserControl
    Implements IXCustomControl

    Dim m_Value As Object

    Public Property Value As Object Implements IXCustomControl.Value
        Get
            Return m_Value
        End Get
        Set(value As Object)
            m_Value = value
            RaiseEvent ValueChanged(Me, value)
        End Set
    End Property

    Public Event ValueChanged As CustomControlValueChangedDelegate Implements IXCustomControl.ValueChanged

End Class

File: FormsAndWpfControls/cs/WinFormsUserControl.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Xarial.XCad.Base.Attributes;
using FormsAndWpfControls.Properties;
using Xarial.XCad.UI.PropertyPage;

namespace FormsAndWpfControls
{
    [Icon(typeof(Resources), nameof(Resources.winforms_icon))]
    public partial class WinFormsUserControl : UserControl, IXCustomControl
    {
        public WinFormsUserControl()
        {
            InitializeComponent();
        }

        public object Value { get; set; }

        public event CustomControlValueChangedDelegate ValueChanged;
    }
}
